core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -217,7 +219,8 @@ class ReplicaManager(val config: KafkaConfig,
 
   def startup() {
     // start ISR expiration thread
-    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs, unit = TimeUnit.MILLISECONDS)
+    // A follower can lag behind leader for up to config.replicaLagTimeMaxMs x (1 + 20%) before it is removed from ISR
+    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs / 5, unit = TimeUnit.MILLISECONDS)
```

Comment:
Sure. I will change this to /2.

On the other hand, I am not sure it is is expensive. When ISR of most partitions don't have to be shrinked, the cost of maybeShrinkIsr() is comparing the latchCatchUpTimeMs of all replicas with current time. Would this cost still be a concern if all these are done in memory without system call? I can update the patch to call time.milliseconds() only once.